src/qml/CoordinateLocator.qml

```diff
@@ -239,4 +308,132 @@ Item {
   {
     flashAnimation.start()
   }
+
+
+  /**
+   * Computes the possible common angle
+   *
+   * @param {QPointF} currentPoint - the current point being proposed
+   * @param {Rubberband} rubberbandModel - holds all previously added points in the current digitizing session
+   * @param {number} commonAngleStepDeg - the common angle in degrees we are trying to snap to
+   * @param {boolean} isRelativeAngle - whether the angle should be calculated relative to the last geometry segment
+   * @returns {number|undefined} - the closest common angle. Returns`undefined` if there is no nearby point that forms a common angle.
+   */
+  function getCommonAngleInDegrees(currentPoint, rubberbandModel, commonAngleStepDeg, isRelativeAngle) {
+    if (!rubberbandModel) {
+      return;
+    }
+
+    const MINIMAL_PIXEL_DISTANCE_TRESHOLD = 20;
+    const SOFT_CONSTRAINT_TOLERANCE_DEGREES = 20;
+    const SOFT_CONSTRAINT_TOLERANCE_PIXEL = 40;
+    const previousPoint = mapCanvas.mapSettings.coordinateToScreen( rubberbandModel.lastCoordinate );
+    const rubberbandPointsCount = rubberbandModel.vertexCount;
+
+    // we need at least 1 alredy digitized point (and the other one is wanna be digitized)
+    if (rubberbandPointsCount < 2) {
+      return;
+    }
+
+    const distanceFromLastPoint = Math.sqrt( (currentPoint.x - previousPoint.x ) ** 2 + ( currentPoint.y - previousPoint.y ) ** 2 );
+    if ( distanceFromLastPoint < MINIMAL_PIXEL_DISTANCE_TRESHOLD ) {
+      return;
+    }
+
+    const commonAngle = commonAngleStepDeg * Math.PI / 180;
+    // see if soft common angle constraint should be performed
+    // only if not in HardLock mode
+    let softAngle = Math.atan2(
+      currentPoint.y - previousPoint.y,
+      currentPoint.x - previousPoint.x
+    );
+    let deltaAngle = 0;
+    if ( isRelativeAngle && rubberbandPointsCount >= 3 ) {
+      // compute the angle relative to the last segment (0° is aligned with last segment)
+      const penultimatePoint = mapCanvas.mapSettings.coordinateToScreen( rubberbandModel.penultimateCoordinate );
```

Comment:
On the formatting front, _I think_ most of the QML javascript code we have doesn't add a space next to the parenthesis the way we do in C++. I.e., in javascript, we (try to) do `if (true)` instead of `if ( true )`.

I don't think we have to settle this now, just pointing it out.

Reply:
Will accomodate, thanks for noting!